internal/runtime/client/test/fake_client.go

```diff
@@ -0,0 +1,149 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package test is used to help with testing functions that need a fake RuntimeClient.
+package test
```

Comment:
```suggestion
package fake
```
nit: afaik it's a convention to call packages like this fake (controller-runtime fake client, client-go)

Reply:
Sounds good 👍🏼